### PR TITLE
Part B: setUpTestData refactor, TransactionTestCase fix, lazy ML imports

### PR DESCRIPTION
**Pull request by brunoamaral**
## Summary

Follow-up to #796 (Part A — config wins, already merged). This is Part B of the CI-speed-up effort: the targeted test-code refactor.

- **B1** — Converted `setUp` → `setUpTestData` in the plan's 10 heaviest-churn test files (plus `test_feedreader_articles.py`, folded in with B3). Real per-file review, not mechanical: only converted classes with genuinely shared, read-only-across-the-class fixtures and more than one test method (a single-test class gets zero speed benefit from `setUpTestData`, so those were left alone to avoid diff noise). Found one non-obvious constraint along the way: `AnnouncementAdmin` instances aren't deep-copyable (they hold a reference to the admin site registry — `TypeError: cannot pickle 'module' object`), so those stay in `setUp`; same for `Client()`/`APIClient()`, which should get fresh login state per test rather than being shared.
- **B3** — Investigated the sole `TransactionTestCase` usage (4 classes in `test_feedreader_articles.py`, which also had the heaviest single-file `setUp` cost). Found no `on_commit` callbacks, no IntegrityError-continuation pattern, and no threading in the command under test — nothing that needs real commits or the truncate-between-tests semantics `TransactionTestCase` pays for. Converted all 4 to `TestCase`; the file's 52 tests still pass.
- **B2** — `gregory/ml/__init__.py` now resolves `get_trainer`/`pseudo.*` via a PEP 562 `__getattr__` instead of eagerly importing `trainer.py` (which imports all three wrappers → tensorflow + torch + transformers + lightgbm) at package-import time. A bare `import gregory.ml`, or importing one wrapper submodule directly, no longer drags in the other two frameworks. `train_models.py`'s top-level `from gregory.ml import get_trainer` is unaffected (it genuinely needs all three).

## Test plan
- [x] Each B1 file's tests run immediately after converting it, individually.
- [x] `test_feedreader_articles.py`'s 52 tests pass both after the `TransactionTestCase→TestCase` conversion and after the subsequent `setUpTestData` refactor.
- [x] ML test files (`test_ml_factory.py`, `test_train_models.py`, `test_predict_articles*.py`, `test_pseudo.py`, `test_summariser.py`, `test_bert_wrapper.py`) pass after the lazy-import change; verified directly that `import gregory.ml` no longer loads tensorflow until an attribute is actually accessed.
- [x] Full suite green: 2698 passed, ~87s (down from ~97-99s after Part A alone).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: django/gregory/ml/__init__.py [hunk 1]
```diff
@@ -6,11 +6,27 @@
...
+
+
+def __getattr__(name):
+	module_name = _LAZY_ATTRS.get(name)
+	if module_name is None:
+		raise AttributeError(f"module {__name__!r} has no attribute {name!r}")
+	import importlib
+
+	module = importlib.import_module(module_name)
+	return getattr(module, name)
```
[the remark this answers is not in the document]
Comment (brunoamaral) [reply django/gregory/ml/__init__.py]: Fixed: `__getattr__` now stores the resolved value in `globals()` before returning, so subsequent plain attribute access hits it directly instead of re-running the lazy import + getattr each time.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: django/subscriptions/tests/test_announcements.py [hunk 2]
```diff
@@ -336,33 +353,37 @@ def test_preheader_fallback_not_shown_when_custom_set(self):
...
 
-		self.lst = Lists.objects.create(
-			list_name="Val List", team=self.team, site=self.site
+		cls.lst = Lists.objects.create(
+			list_name="Val List", team=cls.team, site=cls.site
 		)
 
+	def setUp(self):
 		self.client = Client()
 		self.client.force_login(self.superuser)
```
[the remark this answers is not in the document]
Comment (brunoamaral) [reply django/subscriptions/tests/test_announcements.py:389]: Checked, and this doesn't reproduce: Django's TestCase wraps setUpTestData class attributes in a `TestData` descriptor (django/test/testcases.py, `TestCase.setUpClass`) whose `__get__` deep-copies the value fresh on every `self.cs` access from a new test instance — it isn't the same live object shared across test methods. Verified empirically too: `test_send_test_blocks_when_api_domain_blank` (blanks + saves) and `test_send_test_blocks_when_custom_settings_missing` (deletes) both run alphabetically before `test_send_test_succeeds_with_valid_config` in this file, and the latter still passes with a fresh, valid `cs` — confirming no stale in-memory state leaks between tests. No change made.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_
